collector/internal/extensionapi/client.go

```diff
@@ -180,7 +180,7 @@ func (e *Client) doRequest(req *http.Request, out interface{}) (*http.Response,
 	if err != nil {
 		return nil, err
 	}
-	if resp.StatusCode != 200 {
+	if resp.StatusCode < 200 && resp.StatusCode > 299 {
```

Comment:
I don't recall if this has existing unit tests.  If it does, please consider adding a few test cases for this.

Reply:
This must be `resp.StatusCode < 200 || resp.StatusCode > 299` (Replace `&&` with `||`). Because there is no number `< 200` and `> 299`

Reply:
you're completely right, made that change in a hurry :see_no_evil: 

I didn't find any test for the `extensionapi` package that's why I didn't add any. If you'd like I try adding tests to the package

Reply:
haha, oops... good catch @serkan-ozal.  I missed that too.

Reply:
@serkan-ozal what do you think?  Would you like to see a test here?

Reply:
@tylerbenson I am OK with the PR as is. Of course it would be better if there was a test for this PR. But, since we don't have any test infrastructure for the Lambda collector yet, I don't think missing test should be a blocker.